README.md

```diff
@@ -2,34 +2,45 @@
+
+* `description` (Table) the description is a table that includes the following nested fields:
+	* `summary` (String) a short description of the package, typically less than 100 character long.
+	* `detailed` (String) a long-form description of the package, this should convey the package's principal functionality to the user without being as detailed as the package readme.
+	* `homepage` (String) This is the homepage of the package, which in most cases will be the GitHub URL.
+	* `license` (String) This is [SPDX](https://spdx.org/licenses/) license identifier. Dual licensing is indicated via joining the relevant licenses via `/`.
 
 * `dependencies`
-  * `version` Taken from rockspec. Accepted operators are the relational operators of Lua: == \~= < > <= >= , as well as a special operator, \~>, inspired by the "pessimistic operator" of RubyGems ("\~> 2" means ">= 2, < 3"; "~> 2.4" means ">= 2.4, < 2.5"). No operator means an implicit == (i.e., "lfs 1.0" is the same as "lfs == 1.0"). "lua" is an special dependency name; it matches not a rock, but the version of Lua in use. Multiple version constraints can be joined with a `comma`, e.g. `"neovim >= 5.0, < 7.0"`.
-  * `source`: The source of the dependency. See previous `source` description.
+  * `version` (String) The version constraints on the package.
+    * Accepted operators are the relational operators of Lua: == \~= < > <= >= , as well as a special operator, \~>, inspired by the "pessimistic operator" of RubyGems ("\~> 2" means ">= 2, < 3"; "~> 2.4" means ">= 2.4, < 2.5"). No operator means an implicit == (i.e., "lfs 1.0" is the same as "lfs == 1.0"). "lua" is an special dependency name; it matches not a rock, but the version of Lua in use. Multiple version constraints can be joined with a `comma`, e.g. `"neovim >= 5.0, < 7.0"`.
+    * If no version is specified, then HEAD is assumed valid. 
+    * If no upper bound is specified, then any commit after the tag corresponding to the lower bound is assumed valid. The commit chosen is up to the plugin manager's discretion, but implementers are strongly encouraged to always use the latest valid commit.
+    * If an upper bound is specified, then the the tag corresponding to that upper bound is the latest commit that is valid
+
+  * `source` (String|Table) The source of the dependency. See previous `source` description.
+  * `releases_only` (Boolean) Whether the package manager should only resolve version constraints to include tagged releases.
+
 
-* `external_dependencies` : Like dependencies, this specifies packages which are required for the package but should *not* be managed by the neovim package manager, such as `gcc` or `cmake`
-  * `version` same as `dependencies`
+* `external_dependencies` (Table) Like dependencies, this specifies packages which are required for the package but should *not* be managed by the neovim package manager, such as `gcc` or `cmake`
+  * `version` (String) same as `dependencies`
```

Comment:
Do we want to comment on the obligation of a plugin manager to check external dependency versions?

Reply:
Hmm maybe we should have a field that says whether they are required are not? Some packages (telescope-fzf-native) require gcc at build time, whereas somthing like a python repl requires python at runtime. The latter should maybe be a notification (in the event python is not installed) but the former IMO should be a more immediate error/feedback, since the package will fail to install correctly.

Reply:
That seems worth adding. We still might want to write something about whether or not plugin managers need to check for specific versions of external dependencies - if the dependency in question doesn't provide a clean `--version` command, then the plugin manager may end up needing to maintain a set of `--version` string parsers/similar.